src/backend/distributed/utils/maintenanced.c

```diff
@@ -257,6 +263,13 @@ CitusMaintenanceDaemonMain(Datum main_arg)
 		CHECK_FOR_INTERRUPTS();
 
 		/*
+		 * We clear the metadata cache before every iteration because otherwise
+		 * it might contain stale OIDs if deadlock detection runs during a DROP
+		 * EXTENSION followed by a CREATE EXTENSION.
+		 */
+		ClearMetadataCache();
```

Comment:
> We realized this while running consecutive DROP extension & CREATE extension commands as here, we'd see some ERROR: could not open relation with OID 449464 kind of errors

Right. But that's just hiding a bug.  In fact, I can reproduce at least one bug around this without maintenanced active (opening PR in a couple minutes).